### PR TITLE
Config args

### PR DESCRIPTION
**Pull request by Kotarski**
Test command (this will currently just print out the config):
```
bw wf --project ex --target top lint
```

Added lots of stuff for parsing config files (hierarchically from site->project->target)

Created workflow registries, and dummy lint workflow

### CODE REVIEW

File: blockwork/config/blockwork.py [hunk 1]
```diff
@@ -18,13 +18,16 @@
 @dataclass
 class Blockwork:
     project      : str
+    site         : str                 = "./site.yml"
```
Comment (Intuity) [inline blockwork/config/blockwork.py:21]: In the docs everything so far has been `.yaml` not `.yml` - we should be consistent, I don't mind which way

File: blockwork/common/yaml/converters.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    Creates an object with which to register converters which
+    can be used to initialise a Parser object.
+    """
+    def __init__(self):
+        self._registered_tags: set[str] = set()
+        self._registered_typs: set[Any] = set()
+        self._registry: list[tuple[str, Any, type[Converter]]] = []
+
+    def __iter__(self):
+        for (tag, typ, Converter) in self._registry:
```
[the remark this answers is not in the document]
Comment (Kotarski) [reply blockwork/common/yaml/converters.py:116]: ah good, forgot about yield from

File: blockwork/common/yaml/converters.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        return self.construct_node(loader, node)
+
+    def construct_node(self, loader: Loader, node: yaml.Node) -> _Convertable:
+        raise NotImplementedError
+
+    def represent_node(self, dumper: Dumper, value: _Convertable) -> yaml.Node:
+        raise NotImplementedError
+
+
+class Registry:
```
[the remark this answers is not in the document]
Comment (Kotarski) [reply blockwork/common/yaml/converters.py:105]: IMO no, or at-least not problematically

File: blockwork/activities/workflow.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+@click.pass_obj
+def workflow(ctx : Context, project: str, target: str, workflow_name: str) -> None:
+    """ Run a workflow """
+    wf = cast(type[Workflow], Workflow.get_by_name(workflow_name))
+    wf_types = wf.__call__.__annotations__
+
+    if not ('site' in wf_types, 'project' in wf_types, 'target' in wf_types):
+        raise RuntimeError("Workflow __call__ must specify types for configuration")
+
+    site_parser = parser.Site()
```
[the remark this answers is not in the document]
Comment (Kotarski) [reply blockwork/activities/workflow.py:37]: Maybe... This part is very slapdash at the moment and will definitely be re-written so I'll leave it for now if that's ok

File: blockwork/activities/workflow.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    site_path = ctx.site
+    Site = site_parser(wf_types["site"]).parse(site_path)
+
+    project_parser = parser.Project(Site)
+    project_path = Path(Site.projects[project])
+    Project = project_parser(wf_types["project"]).parse(project_path)
```
[the remark this answers is not in the document]
Comment (Kotarski) [reply blockwork/activities/workflow.py:43]: ditto

File: blockwork/activities/workflow.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    target_parser = parser.Element(Site, Project)
+    target_path = Path(Project.targets[target])
+    Target = target_parser(wf_types["target"]).parse(target_path)
```
[the remark this answers is not in the document]
Comment (Kotarski) [reply blockwork/activities/workflow.py:47]: ditto

**Review by Intuity — APPROVED** [review]

**Pull request merged**